dask_jobqueue/sge.py

```diff
@@ -82,3 +84,42 @@ def __init__(self, queue=None, project=None, resource_spec=None, walltime=None,
         self.job_header = header_template % config
 
         logger.debug("Job script: \n %s" % self.job_script())
+
+    def start_workers(self, n=1, ta=True):
+        """
+        Start workers as a task array
+
+        Parameters
+        ----------
+        n : int
+            Total number of workers to start
+        ta : bool
+            If true, try to use task arrays
+        """
+        logger.debug('starting %s workers', n)
+
+        # Check if Task arrays can be used
+        ta = ta and bool(os.getenv('SGE_TASK_ID', None))
+
+        if ta:
+            logging.debug('Submission using task array')
+            _original_command_template = self._command_template
+            self._command_template = self._command_template.replace('${JOB_ID}', '${JOB_ID}.${SGE_TASK_ID}')
```

Comment:
Maybe we should always have an ID that takes nito account potential job arrays or task arrays. the task array part would potentially be empty.

Reply:
It would be excellent, I am afraid it's difficult to abstract it for all schedulers. I don't remember how job arrays are defined in different schedulers.
Will consider and come up with a suggestion.

Reply:
Hi @guillaumeeb ,
thanks for the hint!! I used your approach, and redefined JOB_ID to handle task arrays.
One caveat to note: `JOB_ID` is the variable name that UGE/SGE uses to define their Job ID. We are kinda breaking the consistency with the scheduler.

In other schedulers, as PBS, the same variable is called `PBS_JOBID`, hence there's no problem...

I don't see it as a problem, but it's worth to keep it in mind.

Reply:
Yep, maybe we would need to change the name at one point.